### PR TITLE
bgp: Add support for BGP Role capabilty and OTC attribute

### PR DESCRIPTION
**Pull request by ton31337**
Signed-off-by: Donatas Abraitis <donatas@opensourcerouting.org>

### CODE REVIEW

File: print-bgp.c [hunk 1]
```diff
@@ -2654,6 +2673,16 @@ bgp_attr_print(netdissect_options *ndo,
         }
         break;
     }
+    case BGPTYPE_OTC:
+    {
+        if (len < 4) {
+            ND_PRINT("invalid len");
+            break;
+        }
+        ND_TCHECK_LEN(pptr, len);
```
Comment (fenner) [inline print-bgp.c:2682]: You don't need the TCHECK since you're using GET_ to get the U4 below.

File: print-bgp.c [hunk 2]
```diff
@@ -2766,6 +2795,15 @@ bgp_capabilities_print(netdissect_options *ndo,
                 cap_offset += 4;
             }
             break;
+        case BGP_CAPCODE_ROLE:
+            if (cap_len < 1) {
+                ND_PRINT(" (too short, < 1)");
+                return;
+            }
+            ND_PRINT("\n\t\t Role name %s",
```
Comment (fenner) [inline print-bgp.c:2803]: Comparing to other capabilities, there's no space between the `\n\t\t` and the description.  Note the additional indentation of the "Role name" vs. "AFI" in the test output:
```
	    Option Capabilities Advertisement (2), length: 3
	      Role Capability (9), length: 1
		 Role name Customer
	    Option Capabilities Advertisement (2), length: 6
	      Multiple Paths (69), length: 4
		AFI IPv4 (1), SAFI Unicast (1), Send/Receive: Receive
```

File: print-bgp.c [hunk 3]
```diff
@@ -2766,6 +2795,15 @@ bgp_capabilities_print(netdissect_options *ndo,
...
+                     tok2str(bgp_role_values, "Unassigned",
```
Comment (fenner) [inline print-bgp.c:2804]: Do you want to use something like "Unassigned (%d)" so that we know what the numeric value was?

**Review by fenner — CHANGES_REQUESTED** [review]
This looks like a good implementation of the new code points in RFC9234.  I have only a few small changes to suggest but otherwise think these changes are ready to merge.

File: print-bgp.c (hunk 3 above)
Comment (ton31337) [reply print-bgp.c:2803]: Thanks, fixed.

File: print-bgp.c (hunk 1 above)
Comment (ton31337) [reply print-bgp.c]: Thanks, fixed.

File: print-bgp.c (hunk 2 above)
Comment (ton31337) [reply print-bgp.c]: Thanks, fixed.

**Comment by fxlb** [comment]
@ton31337 Thank you for the change.
@fenner Thank you for the review.

**Pull request merged**